activerecord/test/cases/associations/has_one_through_disable_joins_associations_test.rb

```diff
@@ -0,0 +1,416 @@
+  end
+
+  def test_preloading_has_one_through_on_belongs_to
+    MemberDetail.delete_all
+    assert_not_nil @member.member_type
+    @organization = organizations(:nsa)
+    @member_detail = MemberDetail.new
+    @member.member_detail = @member_detail
+    @member.organization_without_joins = @organization
+    @member_details = assert_queries(3) do
+      MemberDetail.all.merge!(includes: :member_type_without_joins).to_a
+    end
+    @new_detail = @member_details[0]
+    assert_predicate @new_detail.send(:association, :member_type_without_joins), :loaded?
+    assert_no_queries { @new_detail.member_type_without_joins }
+  end
+
+  def test_save_of_record_with_loaded_has_one_through
+    club = @member.club_without_joins
+    assert_not_nil club.sponsored_member_without_joins
+
+    assert_nothing_raised do
+      Club.find(club.id).save!
+      Club.all.merge!(includes: :sponsored_member_without_joins).find(club.id).save!
+    end
+
+    club.sponsor.destroy
+
+    assert_nothing_raised do
+      Club.find(club.id).save!
+      Club.all.merge!(includes: :sponsored_member_without_joins).find(club.id).save!
+    end
+  end
+
+  def test_through_belongs_to_after_destroy
+    @member_detail = MemberDetail.new(extra_data: "Extra")
+    @member.member_detail = @member_detail
+    @member.save!
+
+    assert_not_nil @member_detail.member_type_without_joins
+    @member_detail.destroy
+    assert_queries(2) do
+      @member_detail.association(:member_type_without_joins).reload
+      assert_not_nil @member_detail.member_type_without_joins
+    end
+
+    @member_detail.member.destroy
+    assert_queries(2) do
+      @member_detail.association(:member_type_without_joins).reload
+      assert_nil @member_detail.member_type_without_joins
+    end
+  end
+
+  def test_value_is_properly_quoted
+    minivan = Minivan.find("m1")
+    assert_nothing_raised do
+      minivan.dashboard_without_joins
+    end
+  end
+
+  def test_has_one_through_polymorphic_with_primary_key_option
+    assert_equal categories(:general), authors(:david).essay_category_without_joins
+
+    authors = Author.joins(:essay_category_without_joins).where("categories.id" => categories(:general).id)
+    assert_equal authors(:david), authors.first
+
+    assert_equal owners(:blackbeard), authors(:david).essay_owner_without_joins
+
+    authors = Author.joins(:essay_owner_without_joins).where("owners.name = 'blackbeard'")
+    assert_equal authors(:david), authors.first
+  end
+
+  def test_has_one_through_with_primary_key_option
+    assert_equal categories(:general), authors(:david).essay_category_2_without_joins
+
+    authors = Author.joins(:essay_category_2_without_joins).where("categories.id" => categories(:general).id)
+    assert_equal authors(:david), authors.first
+  end
+
+  def test_has_one_through_with_default_scope_on_join_model
+    assert_equal posts(:welcome).comments.order("id").first, authors(:david).comment_on_first_post_without_joins
```

Comment:
This one fails locally and again makes me wonder if it's not identifying a bug with the implementation, because it complains `posts` doesn't have an `id` field, which it definitely does.

```
ActiveRecord::StatementInvalid: SQLite3::SQLException: no such column: posts.id
```